server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala

```diff
@@ -753,6 +753,528 @@ trait EmailSetMethodContract {
+           |            "description": null
+           |          }
+           |        }
+           |      }, "c1"]
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  //TODO fix null description
+  def emailSetMailboxIdsResetSuccessAndFailureCanBeMixed(server: GuiceJamesServer): Unit = {
+    val mailboxProbe = server.getProbe(classOf[MailboxProbeImpl])
+    mailboxProbe.createMailbox(MailboxPath.inbox(BOB))
+
+    val path = MailboxPath.forUser(ANDRE, "other")
+    val mailboxId2: MailboxId = mailboxProbe.createMailbox(path)
+
+    val messageId1: MessageId = mailboxProbe
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB),
+        AppendCommand.from(
+          buildTestMessage))
+      .getMessageId
+
+    val messageId2: MessageId = mailboxProbe
+      .appendMessage(BOB.asString, MailboxPath.inbox(BOB),
+        AppendCommand.from(
+          buildTestMessage))
+      .getMessageId
+
+    server.getProbe(classOf[ACLProbeImpl])
+      .replaceRights(path, BOB.asString, new MailboxACL.Rfc4314Rights(Right.Read, Right.Insert))
+
+    val request =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail"],
+         |  "methodCalls": [
+         |    ["Email/set", {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "update": {
+         |        "${messageId1.serialize}": {
+         |          "mailboxIds": {
+         |            "${mailboxId2.serialize}": true
+         |          }
+         |        },
+         |        "${messageId2.serialize}": {
+         |          "mailboxIds": {
+         |            "invalid": true
+         |          }
+         |        }
+         |      }
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .withOptions(new Options(Option.IGNORING_ARRAY_ORDER))
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "75128aab4b1b",
+           |    "methodResponses": [
+           |      ["Email/set", {
+           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |        "newState": "000001",
+           |        "updated": {
+           |          "${messageId1.serialize}": null
+           |        },
+           |        "notUpdated": {
+           |          "${messageId2.serialize}": {
+           |            "type": "invalidArguments",
+           |            "description": null
+           |          }
+           |        }
+           |      }, "c1"]
+           |    ]
+           |}""".stripMargin)
+  }
+
   private def buildTestMessage = {
     Message.Builder
       .of
```

Comment:
I miss the following tests regarding delegation:

```
GIVEN message1 in mailbox AndreMailbox
AND Bob has no rights on AndreMailbox
WHEN Bob tries to set mailboxIds
THEN the operation fails
```

(cannot modify messages I have no rights to - I dsuspect you only tests "no rights" for the target mailbox, not the source one)

```
GIVEN message1 in mailbox AndreMailbox
AND Bob has READ rights on AndreMailbox
WHEN Bob tries to set mailboxIds to [AndreMailbox, bobInbox]
THEN the operation succeeds
```

(read right is enough for a copy)

```
GIVEN message1 in mailbox AndreMailbox
AND Bob has READ rights on AndreMailbox
WHEN Bob tries to set mailboxIds to [bobInbox]
THEN the operation fails
```

(I cannot delete messages from a shared mailbox)

```
GIVEN message1 in mailbox AndreMailbox
AND Bob has READ rights on AndreMailbox
AND Andre has no rights on bobInbox
AND Bob tries to set mailboxIds to [AndreMailbox, bobInbox]
WHEN Andre gets message1
THEN mailboxIds field contains only [AndreMailbox]
```

(the mailboxIds are filtered to the mailboxes the user can access)